File: app/src/main/java/com/brainwellnessspa/MembershipModule/Adapters/MembershipPlanAdapter.java
package com.brainwellnessspa.MembershipModule.Adapters;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.brainwellnessspa.BWSApplication;
import com.brainwellnessspa.MembershipModule.Activities.OrderSummaryActivity;
import com.brainwellnessspa.MembershipModule.Models.MembershipPlanListModel;
import com.brainwellnessspa.R;
import com.brainwellnessspa.databinding.MembershipPlanBinding;

import java.util.ArrayList;

public class MembershipPlanAdapter extends RecyclerView.Adapter<MembershipPlanAdapter.MyViewHolder> {
    private ArrayList<MembershipPlanListModel.Plan> listModelList;
    Context ctx;
    private int row_index = -1, pos = 0;
    Button btnFreeJoin;
    String TrialPeriod;
    Intent i;
    Activity activity;
    public static String planFlag = "", planId = "", price = "";

    public MembershipPlanAdapter(ArrayList<MembershipPlanListModel.Plan> listModelList, Context ctx, Button btnFreeJoin, String TrialPeriod, Activity activity) {
        this.listModelList = listModelList;
        this.ctx = ctx;
        this.TrialPeriod = TrialPeriod;
        this.btnFreeJoin = btnFreeJoin;
        this.activity = activity;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        MembershipPlanBinding v = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext())
                , R.layout.membership_plan, parent, false);
        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        MembershipPlanListModel.Plan listModel = listModelList.get(position);
//        holder.binding.tvTitle.setText(listModel.getTitle());

        holder.binding.tvPlanFeatures01.setText(listModel.getPlanFeatures().get(0).getFeature());
        holder.binding.tvPlanFeatures02.setText(listModel.getPlanFeatures().get(1).getFeature());
        holder.binding.tvPlanFeatures03.setText(listModel.getPlanFeatures().get(2).getFeature());
        holder.binding.tvPlanFeatures04.setText(listModel.getPlanFeatures().get(3).getFeature());
        holder.binding.tvPlanAmount.setText("$" + listModel.getPlanAmount());
        holder.binding.tvSubName.setText(listModel.getSubName());
        holder.binding.tvPlanInterval.setText(listModel.getPlanInterval());

        if (listModel.getRecommendedFlag().equalsIgnoreCase("1")) {
            holder.binding.tvRecommended.setVisibility(View.VISIBLE);
          /*  if (pos == 0) {
                holder.binding.llPlanSub.setBackgroundColor(ctx.getResources().getColor(R.color.blue));
                holder.binding.llFeatures.setVisibility(View.VISIBLE);
                holder.binding.tvPlanAmount.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.tvSubName.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.tvPlanInterval.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.llFeatures.setBackgroundColor(ctx.getResources().getColor(R.color.white));
            }*/
        } else {
            holder.binding.tvRecommended.setVisibility(View.GONE);
        }

        holder.binding.llPlanMain.setOnClickListener(view -> {
            row_index = position;
            pos++;
            notifyDataSetChanged();
        });

        if (row_index == position) {
            ChangeFunction(holder, listModel, position);
        } else {
            if (listModel.getRecommendedFlag().equalsIgnoreCase("1") && pos == 0) {
                holder.binding.tvRecommended.setVisibility(View.VISIBLE);
                ChangeFunction(holder, listModel, position);
            } else {
                holder.binding.llPlanSub.setBackground(ctx.getResources().getDrawable(R.drawable.rounded_light_gray));
                holder.binding.tvPlanAmount.setTextColor(ctx.getResources().getColor(R.color.black));
                holder.binding.tvSubName.setTextColor(ctx.getResources().getColor(R.color.black));
                holder.binding.tvPlanInterval.setTextColor(ctx.getResources().getColor(R.color.black));
                holder.binding.llFeatures.setVisibility(View.GONE);
            }
        }
        btnFreeJoin.setOnClickListener(view -> {
            if (BWSApplication.isNetworkConnected(ctx)) {
                activity.startActivity(i);
                activity.finish();
            } else {
                BWSApplication.showToast(ctx.getString(R.string.no_server_found), activity);
            }
        });
    }

    private void ChangeFunction(MyViewHolder holder, MembershipPlanListModel.Plan listModel, int position) {
        holder.binding.llPlanSub.setBackgroundResource(R.drawable.top_round_blue_cornor);
        holder.binding.llFeatures.setVisibility(View.VISIBLE);
        holder.binding.tvPlanAmount.setTextColor(ctx.getResources().getColor(R.color.white));
        holder.binding.tvSubName.setTextColor(ctx.getResources().getColor(R.color.white));
        holder.binding.tvPlanInterval.setTextColor(ctx.getResources().getColor(R.color.white));
        holder.binding.llFeatures.setBackgroundColor(ctx.getResources().getColor(R.color.white));
        planFlag = listModel.getPlanFlag();
        price = listModel.getPlanAmount();
        planId = listModel.getPlanID();
        i = new Intent(ctx, OrderSummaryActivity.class);
        i.putParcelableArrayListExtra("PlanData", listModelList);
        i.putExtra("TrialPeriod", TrialPeriod);
        i.putExtra("position", position);
        i.putExtra("Promocode", "");
    }

    @Override
    public int getItemCount() {
        return listModelList.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        MembershipPlanBinding binding;

        public MyViewHolder(MembershipPlanBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/brainwellnessspa/membershipModule/adapters/MembershipPlanAdapter.kt
package com.brainwellnessspa.membershipModule.adapters

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.core.content.ContextCompat
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.brainwellnessspa.BWSApplication
import com.brainwellnessspa.R
import com.brainwellnessspa.databinding.MembershipPlanBinding
import com.brainwellnessspa.membershipModule.models.MembershipPlanListModel
import java.util.*

//import com.brainwellnessspa.MembershipModule.Activities.OrderSummaryActivity;
class MembershipPlanAdapter(private val listModelList: ArrayList<MembershipPlanListModel.Plan>, var ctx: Context, var btnFreeJoin: Button, var TrialPeriod: String, var activity: Activity, var i: Intent) : RecyclerView.Adapter<MembershipPlanAdapter.MyViewHolder>() {
    private var rowIndex = -1
    private var pos = 0
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val v: MembershipPlanBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.membership_plan, parent, false)
        return MyViewHolder(v)
    }

    @SuppressLint("SetTextI18n") override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val listModel = listModelList[position] //        holder.binding.tvTitle.setText(listModel.getTitle());
        holder.binding.tvPlanFeatures01.text = listModel.planFeatures?.get(0)?.feature
        holder.binding.tvPlanFeatures02.text = listModel.planFeatures?.get(1)?.feature
        holder.binding.tvPlanFeatures03.text = listModel.planFeatures?.get(2)?.feature
        holder.binding.tvPlanFeatures04.text = listModel.planFeatures?.get(3)?.feature
        holder.binding.tvPlanAmount.text = "$" + listModel.planAmount
        holder.binding.tvSubName.text = listModel.subName
        holder.binding.tvPlanInterval.text = listModel.planInterval
        if (listModel.recommendedFlag.equals("1", ignoreCase = true)) {
            holder.binding.tvRecommended.visibility = View.VISIBLE/*  if (pos == 0) {
                holder.binding.llPlanSub.setBackgroundColor(ctx.getResources().getColor(R.color.blue));
                holder.binding.llFeatures.setVisibility(View.VISIBLE);
                holder.binding.tvPlanAmount.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.tvSubName.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.tvPlanInterval.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.llFeatures.setBackgroundColor(ctx.getResources().getColor(R.color.white));
            }*/
        } else {
            holder.binding.tvRecommended.visibility = View.GONE
        }
        holder.binding.llPlanMain.setOnClickListener {
            rowIndex = position
            pos++
            notifyDataSetChanged()
        }
        if (rowIndex == position) {
            changeFunction(holder, listModel, position)
        } else {
            if (listModel.recommendedFlag.equals("1", ignoreCase = true) && pos == 0) {
                holder.binding.tvRecommended.visibility = View.VISIBLE
                changeFunction(holder, listModel, position)
            } else {
                holder.binding.llPlanSub.background = ContextCompat.getDrawable(activity, R.drawable.rounded_light_gray)
                holder.binding.tvPlanAmount.setTextColor(ContextCompat.getColor(activity, R.color.black))
                holder.binding.tvSubName.setTextColor(ContextCompat.getColor(activity, R.color.black))
                holder.binding.tvPlanInterval.setTextColor(ContextCompat.getColor(activity, R.color.black))
                holder.binding.llFeatures.visibility = View.GONE
            }
        }
        btnFreeJoin.setOnClickListener {
            if (BWSApplication.isNetworkConnected(ctx)) {
                activity.startActivity(i)
                activity.finish()
            } else {
                BWSApplication.showToast(ctx.getString(R.string.no_server_found), activity)
            }
        }
    }

    private fun changeFunction(holder: MyViewHolder, listModel: MembershipPlanListModel.Plan, position: Int) {
        holder.binding.llPlanSub.setBackgroundResource(R.drawable.top_round_green_cornor)
        holder.binding.llFeatures.visibility = View.VISIBLE
        holder.binding.tvPlanAmount.setTextColor(ContextCompat.getColor(activity, R.color.white))
        holder.binding.tvSubName.setTextColor(ContextCompat.getColor(activity, R.color.white))
        holder.binding.tvPlanInterval.setTextColor(ContextCompat.getColor(activity, R.color.white))
        holder.binding.llFeatures.setBackgroundColor(ContextCompat.getColor(activity, R.color.white))
        planFlag = listModel.planFlag.toString()
        price = listModel.planAmount.toString()
        planId = listModel.planID.toString() //        i = new Intent(ctx, OrderSummaryActivity.class);
        //        i.putParcelableArrayListExtra("PlanData", listModelList);
        //        i.putExtra("TrialPeriod", TrialPeriod);
        //        i.putExtra("position", position);
        //        i.putExtra("Promocode", "");
    }

    override fun getItemCount(): Int {
        return listModelList.size
    }

    inner class MyViewHolder(var binding: MembershipPlanBinding) : RecyclerView.ViewHolder(binding.root)

    companion object {
        @JvmField var planFlag = ""

        @JvmField var planId = ""

        @JvmField var price = ""
    }
}